### ISSUE TITLE
Something assert by panflute about pandoc table can be False while pandoc is ok about it

### ISSUE THREAD

**Issue by ickc** [sergiocorreia/panflute#30]
I don't have much, and it might actually be a bug in pandoc, see [bug: docx (containing table) to native and docx to markdown then to native is hugely different - Google Groups](https://groups.google.com/forum/#!topic/pandoc-discuss/nDFRvd8tj8k).

So far what I've got in some kind of MWE is this:

```json
[Table [] [AlignDefault,AlignDefault,AlignDefault] [0.0,0.0,0.0]
 [[Para [Str "x",Space,Str "y"]]
 ,[Para [Strong [Emph [Str "a",Space,Str "b"]]]]
 ,[Para [Strong [Emph [Str "Math"]]]]]
 []]
```

If you try `pandoc -t markdown -f native` on this file, pandoc would accept it. But when passing through panflute, it will complains `IndexError: list index out of range`.

But it could well be a bug from pandoc though.

**Comment by ickc** [comment]
See jgm/pandoc#3285. I have a better MWE there now. As far as panflute is concern, the most important question is the last one there. It seems to me it is reasonable to assert all the lists *should* have the some length.

**Comment by sergiocorreia** [comment]
So as I understand, the docx reader saves the row as a header instead as in the body of the table.

I recall you mentioned this is not allowed in Pandoc, no? (nonetheless, it seems to handle it just fine)

If @jgm intends to allow it, then I can push a workaround, but otherwise it might not be worth it.

**Comment by ickc** [comment]
The one I tested before has no body at all, this one has an empty body. But even if it is allowed, it might cause problems like that issue.

But on the other hand, the reason I was using my filter on that file was to try to capture the table before writing to markdown (the csv is kind of a different representation of the ast) so that I can manually edit the table to correct the output.

Probably we should wait to see how that issue unfold.

**Comment by ickc** [comment]
jgm/pandoc#3285 is closed. They fixed the bug in docx reader so that it won't output a 1 row header with empty body.

However, they didn't enforce the "no. of columns" are the same in the type level yet. But `@jgm` has said that "it is safe to assert that the align-list, width-list, header-list, and each of the row-list are all having the same length".

So it is your call whether the assertion in panflute is relaxed to catch some corner cases from pandoc, or leave it as it currently is since the assertion should be true when the pandoc writers/readers behave correctly.

**Comment by ickc** [comment]
Update: I tested on the fixes done in jgm/pandoc#3285, for the case I posted above, they now have an empty header row. i.e. `[]` instead of `[[],[]]`. I will continue to file bug reports over there. But it seems that some reader authors might uses the fact that pandoc is ok about it to write list an empty list rather than a list of n empty list.

**Comment by sergiocorreia** [comment]
Closed at least for now
